src/Workspaces/CSharp/Portable/Classification/ClassificationHelpers.cs

```diff
@@ -25,11 +25,7 @@ internal static class ClassificationHelpers
         /// <returns>The correct syntactic classification for the token.</returns>
         public static string GetClassification(SyntaxToken token)
         {
-            if (token.IsKind(SyntaxKind.DiscardDesignation, SyntaxKind.UnderscoreToken))
```

Comment:
Looks like `UnderscoreToken` is no longer handled here at all?

So there can also be a `DiscardDesignation` _node_ containing an `UnderscoreToken` token, right (as opposed to the `DiscardDesignation` _token_ here)? In which case that `UnderscoreToken` would no longer be explicitly classified as an `Identifier` here... does the right thing end up happening? Maybe this is what that earlier thread with Cyrus and Neal was about, but I'm not positive.

Reply:
@dpoeschl `SyntaxFacts.IsKeywordKind` calls `SyntaxFacts.IsContextualKeyword` which returns `true` for `UnderscoreToken` . https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs#L1112  So both `UnderscoreToken` and `DiscardDesignation` would be classified as Keywords.

Reply:
Okay, then can we have a comment to that effect here that outlines why it's skipping that check? By just reading the code/tests today, I'm having a hard time understanding that this will all come together into a consistent experience through all of the parses/cases. But if I'm missing a good explanation of it somewhere then let me know where that is. :)

Reply:
I added a comment explaining why only DiscardDesignation is needed here.